ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -125,4 +216,191 @@ mod tests {
         };
         check(&nd, &mut rng, 1_u8);
     }
+
+    /// Tests for Geometric
+    #[test]
+    fn test_geometric_constructor() {
+        // should fail for negative probability
+        let p = -1.0;
+        let expected = Err(Error::BadGeometricProb(p));
+        let actual = Geometric::new(p);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_geometric_sample_dist() {
+        let mut rng = rand::thread_rng();
+        let p = 0.5; // success probability
+        let geometric = Geometric::new(p).expect("Geometric not constructed");
+        let mut histogram = HashMap::new();
+        let num_samples = 100_000;
+        for _ in 0..num_samples {
+            let sample = geometric.sample(&mut rng);
+            *histogram.entry(sample).or_insert(0) += 1;
+        }
+        #[allow(clippy::cast_precision_loss)]
+        for x in 0..100 {
+            let observed_probability = histogram
+                .get(&x)
+                .map_or(0.0, |count| f64::from(*count) / f64::from(num_samples));
+            let expected_probability = (1.0 - p).powf(f64::from(x)) * p;
+            println!("x = {x}, Observed Probability = {observed_probability}, Expected Probability = {expected_probability}");
+            assert!((observed_probability - expected_probability) <= 0.01);
+        }
+    }
+    /// Tests for Double Geometric
+    #[test]
+    fn test_double_geometric_constructor() {
+        // should fail for negative s parameter
+        let s = -1.0;
+        let n = 25;
+        let expected = Err(Error::BadS(s));
+        let actual = DoubleGeometric::new(s, n);
+        assert_eq!(expected, actual);
+    }
+    /// Tests for Truncated Double Geometric
+    #[test]
+    fn test_truncated_double_geometric_constructor() {
+        // should fail for negative s parameter
+        let mut s = -1.0;
+        let mut n = 25;
+        let mut expected = Err(Error::BadS(s));
+        let mut actual = TruncatedDoubleGeometric::new(s, n);
+        assert_eq!(expected, actual);
+        s = 2.0;
+        n = 3_000_000;
+        expected = Err(Error::BadShiftValue(n));
+        actual = TruncatedDoubleGeometric::new(s, n);
+        assert_eq!(expected, actual);
+    }
+    #[test]
+    fn test_truncated_double_geometric() {
+        let mut rng = rand::thread_rng();
+        let s = 1.0;
+        let n = 25;
```

Comment:
constants can use const rather than let.

Reply:
but then I would have to make it upper case to make the warning go away... and that would make it look different than the parameters it matches to in the function call.

Reply:
Yes, but that's how you do these things in languages with naming conventions.

The way we generally avoid that style of problem is to give things names rather than relying on letters.

Reply:
alright, that makes sense.  maybe what I'll do here is what I have in the other test functions of testing more combinations of parameters and update `s` and `n`